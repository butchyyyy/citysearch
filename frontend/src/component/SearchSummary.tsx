import City from "model/City"
import React from "react"
import { Badge } from "reactstrap"

interface Props {
  /** Current search result */
  cities: City[]
}

/** Shows summary for current search result as a number of cities */
const SearchSummary = (props: Props) => (
    <h6>Cities: <Badge color="secondary">{props.cities && props.cities.length}</Badge></h6>
)

export default SearchSummary
