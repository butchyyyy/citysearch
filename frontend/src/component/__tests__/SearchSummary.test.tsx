import { testCities } from "__tests__/TestData"
import SearchSummary from "component/SearchSummary"
import { shallow } from "enzyme"
import React from "react"

describe("Test SearchSummmary Component", () => {
  const wrapper = shallow(<SearchSummary cities={testCities} />)

  it("Test SarchSummary Snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
