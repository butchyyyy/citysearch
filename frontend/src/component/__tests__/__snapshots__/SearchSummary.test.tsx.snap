// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchSummmary Component Test SarchSummary Snapshot 1`] = `
<h6>
  Cities: 
  <Badge
    color="secondary"
    pill={false}
    tag="span"
  >
    3
  </Badge>
</h6>
`;
